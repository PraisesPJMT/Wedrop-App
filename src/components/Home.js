import React from 'react';

const Home = () => {
  const title = 'Home Page';
  return (
    <>
      <h1>{title}</h1>
    </>
  );
};

export default Home;
