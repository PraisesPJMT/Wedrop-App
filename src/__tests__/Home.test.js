import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../components/Home';

describe('Home page component tests', () => {
  test('Renders home page component correctly', () => {
    const home = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(home).toMatchSnapshot();
  });

  test('Check for heading \'Wedrop\' to be in the Home page', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(
      document.querySelector('h1').textContent,
    ).toBe('Wadrop');
  });
});
