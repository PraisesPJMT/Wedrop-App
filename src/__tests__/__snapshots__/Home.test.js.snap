// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component tests Renders home page component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="text-white relative"
      >
        <img
          alt="World Map"
          class="w-full md:w-auto opacity-20"
          src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/world-map.png"
        />
        <div
          class="absolute top-[20%] right-6 md:right-[20%] text-right w-3/4"
        >
          <h1
            class="font-black text-2xl"
          >
            Wadrop
          </h1>
          <h2
            class="font-bold text-xm"
          >
            Check Weather in Cities All Over the World
          </h2>
          <p
            class="font-xl"
          >
            22,418
            + Cities
          </p>
        </div>
      </section>
      <div
        class="text-white bg-strip-blue p-1"
      >
        <p
          class="text-xs font-bold"
        >
          COUNTRIES BY CONTINENTS
        </p>
      </div>
      <section
        class="text-white grid grid-cols-2 overflow-y-auto md:grid-cols-6 md:overflow-hidden"
      >
        <a
          class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue "
          href="/africa"
        >
          <img
            alt="Africa"
            class="w-4/5 p-3 opacity-50 hover:scale-110"
            src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/africa.png"
          />
          <p
            class="flex flex-col absolute bottom-3 right-3 text-right"
          >
            <span
              class="font-bold"
            >
              Africa
            </span>
            <span
              class=""
            >
              2,106
              + cities
            </span>
          </p>
        </a>
        <a
          class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue bg-dark-blue"
          href="/asia"
        >
          <img
            alt="Asia"
            class="w-4/5 p-3 opacity-50 hover:scale-110"
            src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/asia.png"
          />
          <p
            class="flex flex-col absolute bottom-3 right-3 text-right"
          >
            <span
              class="font-bold"
            >
              Asia
            </span>
            <span
              class=""
            >
              7,296
              + cities
            </span>
          </p>
        </a>
        <a
          class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue bg-dark-blue"
          href="/europe"
        >
          <img
            alt="Europe"
            class="w-4/5 p-3 opacity-50 hover:scale-110"
            src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/europe.png"
          />
          <p
            class="flex flex-col absolute bottom-3 right-3 text-right"
          >
            <span
              class="font-bold"
            >
              Europe
            </span>
            <span
              class=""
            >
              6,690
              + cities
            </span>
          </p>
        </a>
        <a
          class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue "
          href="/oceania"
        >
          <img
            alt="Oceania"
            class="w-4/5 p-3 opacity-50 hover:scale-110"
            src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/oceania.png"
          />
          <p
            class="flex flex-col absolute bottom-3 right-3 text-right"
          >
            <span
              class="font-bold"
            >
              Oceania
            </span>
            <span
              class=""
            >
              183
              + cities
            </span>
          </p>
        </a>
        <a
          class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue "
          href="/south-america"
        >
          <img
            alt="South America"
            class="w-4/5 p-3 opacity-50 hover:scale-110"
            src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/south-america.png"
          />
          <p
            class="flex flex-col absolute bottom-3 right-3 text-right"
          >
            <span
              class="font-bold"
            >
              South America
            </span>
            <span
              class=""
            >
              2,057
              + cities
            </span>
          </p>
        </a>
        <a
          class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue bg-dark-blue"
          href="/north-america"
        >
          <img
            alt="North America"
            class="w-4/5 p-3 opacity-50 hover:scale-110"
            src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/north-america.png"
          />
          <p
            class="flex flex-col absolute bottom-3 right-3 text-right"
          >
            <span
              class="font-bold"
            >
              North America
            </span>
            <span
              class=""
            >
              4,086
              + cities
            </span>
          </p>
        </a>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="text-white relative"
    >
      <img
        alt="World Map"
        class="w-full md:w-auto opacity-20"
        src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/world-map.png"
      />
      <div
        class="absolute top-[20%] right-6 md:right-[20%] text-right w-3/4"
      >
        <h1
          class="font-black text-2xl"
        >
          Wadrop
        </h1>
        <h2
          class="font-bold text-xm"
        >
          Check Weather in Cities All Over the World
        </h2>
        <p
          class="font-xl"
        >
          22,418
          + Cities
        </p>
      </div>
    </section>
    <div
      class="text-white bg-strip-blue p-1"
    >
      <p
        class="text-xs font-bold"
      >
        COUNTRIES BY CONTINENTS
      </p>
    </div>
    <section
      class="text-white grid grid-cols-2 overflow-y-auto md:grid-cols-6 md:overflow-hidden"
    >
      <a
        class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue "
        href="/africa"
      >
        <img
          alt="Africa"
          class="w-4/5 p-3 opacity-50 hover:scale-110"
          src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/africa.png"
        />
        <p
          class="flex flex-col absolute bottom-3 right-3 text-right"
        >
          <span
            class="font-bold"
          >
            Africa
          </span>
          <span
            class=""
          >
            2,106
            + cities
          </span>
        </p>
      </a>
      <a
        class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue bg-dark-blue"
        href="/asia"
      >
        <img
          alt="Asia"
          class="w-4/5 p-3 opacity-50 hover:scale-110"
          src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/asia.png"
        />
        <p
          class="flex flex-col absolute bottom-3 right-3 text-right"
        >
          <span
            class="font-bold"
          >
            Asia
          </span>
          <span
            class=""
          >
            7,296
            + cities
          </span>
        </p>
      </a>
      <a
        class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue bg-dark-blue"
        href="/europe"
      >
        <img
          alt="Europe"
          class="w-4/5 p-3 opacity-50 hover:scale-110"
          src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/europe.png"
        />
        <p
          class="flex flex-col absolute bottom-3 right-3 text-right"
        >
          <span
            class="font-bold"
          >
            Europe
          </span>
          <span
            class=""
          >
            6,690
            + cities
          </span>
        </p>
      </a>
      <a
        class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue "
        href="/oceania"
      >
        <img
          alt="Oceania"
          class="w-4/5 p-3 opacity-50 hover:scale-110"
          src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/oceania.png"
        />
        <p
          class="flex flex-col absolute bottom-3 right-3 text-right"
        >
          <span
            class="font-bold"
          >
            Oceania
          </span>
          <span
            class=""
          >
            183
            + cities
          </span>
        </p>
      </a>
      <a
        class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue "
        href="/south-america"
      >
        <img
          alt="South America"
          class="w-4/5 p-3 opacity-50 hover:scale-110"
          src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/south-america.png"
        />
        <p
          class="flex flex-col absolute bottom-3 right-3 text-right"
        >
          <span
            class="font-bold"
          >
            South America
          </span>
          <span
            class=""
          >
            2,057
            + cities
          </span>
        </p>
      </a>
      <a
        class="h-[180px] relative md:bg-cornflower-blue md:border md:border-dark-blue bg-dark-blue"
        href="/north-america"
      >
        <img
          alt="North America"
          class="w-4/5 p-3 opacity-50 hover:scale-110"
          src="https://raw.githubusercontent.com/PraisesPJMT/Data/main/maps/north-america.png"
        />
        <p
          class="flex flex-col absolute bottom-3 right-3 text-right"
        >
          <span
            class="font-bold"
          >
            North America
          </span>
          <span
            class=""
          >
            4,086
            + cities
          </span>
        </p>
      </a>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
